Assets/Scripts/CombatZone.cs

```diff
@@ -33,50 +38,97 @@ public void Start()
 
     }
 
+    public void Update()
+    {
+        if (!Timer.IsInProgress())
+            return;
+
+        if (CombatStarted)
+            DisablePlayerMovement();
+
+        if (!Timer.IsFinished())
+            SetCombatPositionsUpdate();
+    }
+
     void OnTriggerEnter(Collider other)
     {
         if (CombatStarted)
             return;
 
         if (other.gameObject.CompareTag("Player"))
         {
+            Timer.StartTimer(MovementTime);
+            StartCoroutine(StartingCombat());
             InitiateCombat();
         }
     }
 
     private void InitiateCombat()
     {
         CombatStarted = true;
+        InitalPositionsEnemies = new Vector3[Enemies.Length];
+        InitalPositionsPlayers = new Vector3[Players.Length];
 
-        SetCombatPositions();
+        for (int i = 0; i < InitalPositionsEnemies.Length; i++)
+        {
+            InitalPositionsEnemies[i] = Enemies[i].transform.position;
+            Enemies[i].GetComponent<Combatant>().EnterCombat();
```

Comment:
these EnterCombat() should not be here. Place them once you actually start combat. Rn it's done before you start moving them, which is weird

Reply:
Fixed! Moved it to the end of the coroutine.